studio/src/mas-placeholders.js

```diff
@@ -0,0 +1,1479 @@
+
+            if (
+                placeholder.key === this.editedKey &&
+                placeholder.value === this.editedValue
+            ) {
+                console.log('No changes detected, skipping save operation');
+                Events.toast.emit({
+                    variant: 'info',
+                    content: 'No changes to save',
+                });
+
+                this.editingPlaceholder = null;
+                this.editedKey = '';
+                this.editedValue = '';
+                this.draftStatus = false;
+
+                return;
+            }
+
+            const fragmentPath = placeholder._fragment?.path;
+            if (!fragmentPath) {
+                throw new Error('Fragment path is missing');
+            }
+
+            let fragmentData = null;
+            try {
+                fragmentData =
+                    await this.repository.getFragmentByPath(fragmentPath);
+            } catch (fetchError) {
+                fragmentData = placeholder._fragment;
+            }
+
+            if (!fragmentData) {
+                throw new Error('Fragment data is missing');
+            }
+
+            const updatedFragment = { ...fragmentData };
+
+            const keyFieldIndex = updatedFragment.fields.findIndex(
+                (field) => field.name === 'key',
+            );
+            if (keyFieldIndex !== -1) {
+                updatedFragment.fields[keyFieldIndex].values = [this.editedKey];
+            } else {
+                updatedFragment.fields.push({
+                    name: 'key',
+                    values: [this.editedKey],
+                });
+            }
+
+            const valueFieldIndex = updatedFragment.fields.findIndex(
+                (field) => field.name === 'value',
+            );
+            if (valueFieldIndex !== -1) {
+                updatedFragment.fields[valueFieldIndex].values = [
+                    this.editedValue,
+                ];
+            } else {
+                updatedFragment.fields.push({
+                    name: 'value',
+                    values: [this.editedValue],
+                });
+            }
+
+            const locReadyFieldIndex = updatedFragment.fields.findIndex(
+                (field) => field.name === 'locReady',
+            );
+            if (locReadyFieldIndex !== -1) {
+                updatedFragment.fields[locReadyFieldIndex].values = [true];
+            } else {
+                updatedFragment.fields.push({
+                    name: 'locReady',
+                    values: [true],
+                });
+            }
+
+            const savedFragment =
+                await this.repository.saveDictionaryFragment(updatedFragment);
+
+            const updatedPlaceholder = {
+                ...placeholder,
+                key: this.editedKey,
+                value: this.editedValue,
+                status: savedFragment.status || 'Draft',
+                _fragment: savedFragment,
```

Comment:
please respect coding convention, we do not start variables with underscore.

Reply:
fixed